templates/pages/assets/customfield.html.twig

```diff
@@ -0,0 +1,174 @@
+    <div class="placeholder-type-info">
+        {{ alerts.alert_info(__('Placeholders are empty fields without a label or value to help with the form layout')) }}
+    </div>
+    {{ fields.nullField() }}
+
+
+    {{ fields.sliderField('field_options[full_width]', item.fields['field_options']['full_width']|default(false), __('Full width'), {
+        add_field_class: 'd-none'
+    }) }}
+    {{ fields.sliderField('field_options[readonly]', item.fields['field_options']['readonly']|default(false), __('Readonly'), {
+        add_field_class: 'd-none'
+    }) }}
+    {{ fields.sliderField('field_options[required]', item.fields['field_options']['required']|default(false), __('Mandatory'), {
+        add_field_class: 'd-none'
+    }) }}
+    {{ fields.sliderField('field_options[multiple]', item.fields['field_options']['multiple']|default(false), __('Multiple values'), {
+        add_field_class: 'd-none'
+    }) }}
+    {{ fields.sliderField('field_options[enable_richtext]', item.fields['field_options']['enable_richtext']|default(false), __('Rich text'), {
+        add_field_class: 'd-none'
+    }) }}
+    {{ fields.numberField('field_options[min]', item.fields['field_options']['min']|default(0), __('Minimum'), {
+        add_field_class: 'd-none'
+    }) }}
+    {{ fields.numberField('field_options[max]', item.fields['field_options']['max']|default(constant('PHP_INT_MAX')), __('Maximum'), {
+        add_field_class: 'd-none'
+    }) }}
+    {{ fields.numberField('field_options[step]', item.fields['field_options']['step']|default(1), _n('Step', 'Steps', 1), {
+        add_field_class: 'd-none',
+        step: 0.01
+    }) }}
+
+    {{ fields.dropdownArrayField('itemtype', item.fields['itemtype']|default('_custom_options'), allowed_dropdown_itemtypes, __('Item type'), {
+        add_field_class: 'd-none'
+    }) }}
+    {{ fields.dropdownArrayField('field_options[values]', '', item.fields['field_options']['values']|default([]), __('Allowed values'), {
+        add_field_class: 'd-none',
+        multiple: true,
+        enable_tags: true,
+        values: item.fields['field_options']['values']|default([])
+    }) }}
+
+    <div class="field-default-value"></div>
+
+    <script>
+        $(() => {
+            const autoUpdateNameField = () => {
+                let name_prefix = '';
+                const selected_type = $('select[name="type"]').val();
+                if (selected_type === 'date' || selected_type === 'datetime') {
+                    name_prefix = 'date_';
+                }
+                // label is made lowercase and spaces are replaced with underscores. All other characters are removed
+                $('input[name="name"]').val(name_prefix + $('input[name="label"]').val().toLowerCase().replace(/ /g, '_').replace(/[^a-z_]/g, ''));
+                $('.form-field.displayed-name .form-control-plaintext').text($('input[name="name"]').val());
+            };
```

Comment:
For the asset definition itself, I decided to not define the system name automatically from the user defined label to prevent the following issues:
 - a label based on non latin chars would result in placeholders chars only (e.g. `رقم ضمان الجودة`, the arabic translation of `QA number`, results in `__` system name);
 - similar labels could result in identical system names (e.g. `ipv4` and `ipv6` both results in `ipv` system name).

IMHO, it is preferable to let the administrator define its own system name. Also, in this first batch, we could probably remove the `label` field. We may reimplement it later with translations support, as we did for the asset definition system name.